src/engine/Service.cpp

```diff
@@ -125,6 +125,31 @@ ProtoResult Service::computeResultImpl([[maybe_unused]] bool requestLaziness) {
   ad_utility::httpUtils::Url serviceUrl{
       asStringViewUnsafe(parsedServiceClause_.serviceIri_.getContent())};
 
+  // Receive updates about the RuntimeInformation from the service endpoint.
+  const std::string queryId = ad_utility::UuidGenerator()();
+  auto updateRuntimeInformation = [&]() {
+    try {
+      const std::string target = absl::StrCat("/watch/", queryId);
+      for (const auto& msg :
+           networkFunctions_.getRuntimeInfoFunction_(serviceUrl, target)) {
```

Comment:
We have to really take care that this doesn't stall forever.
In particular we want to set a timeout/cancellation request on the websocket connection when the `Service` is destroyed. (probably your websocket function needs an explicit mechanism for closing, and be a proper struct and not only a single function.